packages/mds-compliance-api/handlers/get-violation-periods.ts

```diff
@@ -0,0 +1,85 @@
+import { ComplianceServiceClient, ComplianceViolationPeriodDomainModel } from '@mds-core/mds-compliance-service'
+import { ApiRequestQuery } from '@mds-core/mds-api-server'
+import express from 'express'
+import { parseRequest } from '@mds-core/mds-api-helpers'
+import { Timestamp } from '@mds-core/mds-types'
+import { BadParamsError, isDefined, now } from '@mds-core/mds-utils'
+import { ComplianceAggregate, ComplianceApiRequest, ComplianceApiResponse } from '../@types'
+
+export type ComplianceApiGetViolationPeriodsRequest = ComplianceApiRequest &
+  ApiRequestQuery<'start_time' | 'end_time' | 'provider_ids' | 'policy_ids'>
+
+export type ComplianceApiGetViolationPeriodsResponse = ComplianceApiResponse<{
+  start_time: Timestamp
+  end_time: Timestamp
+  results: ComplianceAggregate[]
+}>
+
+function encodeToken(ids: string[]): string {
+  const buffer = Buffer.from(ids.join(','))
+  return buffer.toString('base64')
+}
```

Comment:
It would be nice if the `encodeToken` and `decodeToken` functions weren't defined in separate files.